homework/list.js

```diff
@@ -1,79 +1,99 @@
 'use strict';
+(function(exports) {
 
-(function() {
+  /**
+  * Create a instance of the ListManager.
+  *
+  * @constructor
+  */
+	var ListManager = function() {
+		/** @private */ this._listNoteContent = [];
+		/** @private */ this._wrapper = document.querySelector('#note-list-wrapper');
+		document.addEventListener('DOMContentLoaded', this.start.bind(this));
+	}
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	ListManager.prototype = {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+		/** Initialize the listManager. */
+		start() {
+			this.fetchList()
+				.then(this.updateList.bind(this))
+				.then(this.drawList.bind(this))
+				.then(this.preloadFirstNote.bind(this))
+				.catch(function(error) {
+					console.error(error);
```

Comment:
It's necessary to re-throw the error after logging it. Since you will silent the error after this step.